lib/utils.pm

```diff
@@ -327,8 +327,23 @@ sub unlock_if_encrypted {
         assert_screen("encrypted-disk-password-prompt", 200);
         type_password;    # enter PW at boot
         save_screenshot;
-        assert_screen 'encrypted_disk-typed_password' if $args{check_typed_password};
-        send_key "ret";
+        if ($args{check_typed_password}) {
+            assert_screen([qw(encrypted_disk-typed_password encrypted_disk-typed_password_missing_character)]);
```

Comment:
Hmm, as we only 1 retry to type it, what's about just checking that we didn't type it properly? I would try to avoid adding new needle, as we also do not want to have 8 needles for 1-8 missing characters, which might happen in theory.
Will it be possible?

Reply:
why 8 needles? so far I have noticed only one character missing in failures, so it's just one needle.

Reply:
Yes, but what if we will get the case with 2 missing characters? I've described the worst case of course. But the point is that we can have no extra needles, which will help with maintenance too, as we won't have to re-create them once font changes. So if it's possible to avoid having any new tag/needle, I would highly recommend to do so. And for me it looks like we can avoid it in this particular case.

Reply:
I am not sure, I understand. If I use the existing needles, then I could follow the logic "If the password needle doesn't match, enter and wait for password prompt or delete previously typed password, retype the password, match password and enter", but I don't know if that's what you have in mind. Please, share your insight.

Reply:
Yes, I thought of using logic that if needle doesn't match we retry. We also could improve https://openqa.suse.de/tests/5474636#step/boot_encrypt/1 needle to assert that nothing is entered before typing.

Reply:
I am not sure if changing the needle would be a good idea because there is the cursor blinking which would require more tries to match (not sure if this makes it significantly harder or not) plus, I have set the "erase password" to one extra backspace just in case the backend misses any. If you think that the blinking is insignificant, I will change the needle.

Reply:
We can use `do not match` part to skip the cursor and still ensure that there is nothing typed yet.

Reply:
In that case, if there is one character typed, it will not be seen. I will try to see if I can make the "exclude" area cover only the cursor (so it will still see half the character).

Reply:
AFAR cursor there is horizontal, so we can use this assumption. Even if it's vertical, it's thinner than `*` symbol, so there should be no issue. As a side note, I'm not sure if we need this at all and my original point was that we can retry if needle for the typed password doesn't match. And in case we have typos, we can make this logic more complex.

Reply:
Ok, done. I am running VRs.

Reply:
> And in case we have typos, we can make this logic more complex.

?

Reply:
This solution is not bulletproof as, for instance, we re-try only once. And in case we type 9 characters, we won't even re-try. But let's proceed with what we have, as we don't face those issues yet.